In [1]:
import pandas as pd

from src.features.nhanes_features import (
    create_basic_health_features,
    clean_features
)

ModuleNotFoundError: No module named 'src'

In [2]:
import os

os.getcwd()

'/workspaces/adaptive-health-engine/notebooks'

In [3]:
import sys
import os

sys.path.append(os.path.abspath(".."))

In [5]:
import pandas as pd

from src.features.nhanes_features import (
    create_basic_health_features,
    clean_features
)

In [6]:
import pandas as pd

body = pd.read_sas(
    "../data/raw/nhanes/body.XPT"
)

demo = pd.read_sas(
    "../data/raw/nhanes/demographics.XPT"
)

print(body.shape)
print(demo.shape)

(8860, 22)
(11933, 27)


In [7]:
df = demo.merge(
    body,
    on="SEQN",
    how="inner"
)

print(df.shape)

(8860, 48)


In [8]:
df.columns.tolist()

['SEQN',
 'SDDSRVYR',
 'RIDSTATR',
 'RIAGENDR',
 'RIDAGEYR',
 'RIDAGEMN',
 'RIDRETH1',
 'RIDRETH3',
 'RIDEXMON',
 'RIDEXAGM',
 'DMQMILIZ',
 'DMDBORN4',
 'DMDYRUSR',
 'DMDEDUC2',
 'DMDMARTZ',
 'RIDEXPRG',
 'DMDHHSIZ',
 'DMDHRGND',
 'DMDHRAGZ',
 'DMDHREDZ',
 'DMDHRMAZ',
 'DMDHSEDZ',
 'WTINT2YR',
 'WTMEC2YR',
 'SDMVSTRA',
 'SDMVPSU',
 'INDFMPIR',
 'BMDSTATS',
 'BMXWT',
 'BMIWT',
 'BMXRECUM',
 'BMIRECUM',
 'BMXHEAD',
 'BMIHEAD',
 'BMXHT',
 'BMIHT',
 'BMXBMI',
 'BMDBMIC',
 'BMXLEG',
 'BMILEG',
 'BMXARML',
 'BMIARML',
 'BMXARMC',
 'BMIARMC',
 'BMXWAIST',
 'BMIWAIST',
 'BMXHIP',
 'BMIHIP']

In [9]:
health_features = create_basic_health_features(df)

health_features.head()

,age,weight_kg,height_cm,bmi
0,43.0,86.9,179.5,27.0
1,66.0,101.8,174.2,33.5
2,44.0,69.4,152.9,29.7
3,5.0,34.3,120.1,23.8
4,2.0,13.6,NaN,NaN


In [10]:
health_features["age"].describe()

count    8.860000e+03
mean     3.990542e+01
std      2.517546e+01
min      5.397605e-79
25%      1.500000e+01
50%      4.000000e+01
75%      6.300000e+01
max      8.000000e+01
Name: age, dtype: float64

In [11]:
adult_features = health_features[
    health_features["age"] >= 18
]

adult_features.shape


(6337, 4)

Raw NHANES participants:
8860

After adult filter (age ≥18):
6337 participants

Features:
4
- age
- weight_kg
- height_cm
- bmi

In [12]:
adult_features.isnull().sum()

age            0
weight_kg     89
height_cm     75
bmi          102
dtype: int64

adult_features
6337 participants

Missing values:

age          0
weight_kg   89
height_cm   75
bmi         102

In [13]:
clean_features = adult_features.dropna()

clean_features.shape

(6235, 4)

Adult NHANES dataset

Before cleaning:
6337 participants

After removing missing measurements:
6235 participants

Features:
- age
- weight_kg
- height_cm
- bmi

In [14]:
clean_features.to_csv(
    "../data/processed/nhanes_health_features.csv",
    index=False
)

In [15]:
import os

os.path.exists("../data/processed/nhanes_health_features.csv")

True

In [16]:
import pandas as pd

test_load = pd.read_csv(
    "../data/processed/nhanes_health_features.csv"
)

test_load.head()

,age,weight_kg,height_cm,bmi
0,43.0,86.9,179.5,27.0
1,66.0,101.8,174.2,33.5
2,44.0,69.4,152.9,29.7
3,34.0,90.6,173.3,30.2
4,68.0,103.5,155.9,42.6
